apps/emqx/src/emqx_broker.erl

```diff
@@ -151,13 +154,9 @@ with_subid(undefined, SubOpts) ->
 with_subid(SubId, SubOpts) ->
     maps:put(subid, SubId, SubOpts).
 
-%% @private
-do_subscribe(Topic, SubPid, SubOpts) ->
-    true = ets:insert(?SUBSCRIPTION, {SubPid, Topic}),
-    Group = maps:get(share, SubOpts, undefined),
-    do_subscribe(Group, Topic, SubPid, SubOpts).
-
-do_subscribe(undefined, Topic, SubPid, SubOpts) ->
+do_subscribe(Topic, SubPid, SubOpts) when is_binary(Topic) ->
+    %% FIXME: subscribe shard bug
+    %% https://emqx.atlassian.net/browse/EMQX-10214
```

Comment:
maybe misunderstood? see jira for more comments.

Reply:
@JimMoen  please confirm if there should really be a FIXME. maybe delete the comment if there is already a jira to track it.

Reply:
Need more discussion. The comment will be deleted for now.